service/history/visibilityQueueTaskExecutor.go

```diff
@@ -436,9 +439,34 @@ func (t *visibilityQueueTaskExecutor) processDeleteExecution(
 		StartTime:   task.StartTime,
 		CloseTime:   task.CloseTime,
 	}
+	if t.shard.GetConfig().VisibilityProcessorEnsureCloseBeforeDelete() {
+		if !t.isVisibilityClosed(task) {
+			return consts.ErrTaskRetry
+		}
+	}
 	return t.visibilityMgr.DeleteWorkflowExecution(ctx, request)
 }
 
+func (t *visibilityQueueTaskExecutor) isVisibilityClosed(task *tasks.DeleteExecutionVisibilityTask) bool {
```

Comment:
Since this func is specific for checking if closeExecutionVisibility task is completed or not, I think the function should be renamed to reflect that.

Reply:
Renamed to `isCloseExecutionVisibilityTaskPending` and inverted logic. I think this is clearer, but lmk what you think.